### PR TITLE
Benny/targeted optimistic sync

### PR DESCRIPTION
**Pull request by benny-conn**
Changes:

- **Added** new handler for receiving tokens from alchemy webhooks
- **Fixed** optimistic syncing push notifications (forgot a return statement)

Alchemy webhooks will send a request with as much grouping as possible for all activity for a given contract. Right now we will set it up to just receive prohibition data.

### CODE REVIEW

File: service/notifications/notifications.go [hunk 1]
```diff
@@ -609,6 +609,7 @@ func createPushMessage(ctx context.Context, notif db.Notification, queries *db.Q
 		} else {
 			message.Body = fmt.Sprintf("You received a new %s token", name)
 		}
+		return message, nil
```
Comment (benny-conn) [inline service/notifications/notifications.go:612]: This is all that is needed to have push notifs work for optimistic syncing given that I already set them up earlier.

File: tokenprocessing/process.go [hunk 2]
```diff
@@ -243,6 +246,272 @@ func processOwnersForUserTokens(mc *multichain.Provider, queries *coredb.Queries
...
+				chain = persist.ChainETH
+			case "POLYGON_MAINNET":
+				chain = persist.ChainPolygon
+			case "ARBITRUM_MAINNET":
+				chain = persist.ChainArbitrum
+			case "OPTIMISM_MAINNET":
+				chain = persist.ChainOptimism
+			default:
+				util.ErrResponse(c, http.StatusInternalServerError, fmt.Errorf("invalid alchemy network: %s", input.Event.Network))
+				return
```
Comment (jarrel-b) [inline tokenprocessing/process.go:375]: Would we extend the webhook to allow for other chains in the future? Wonder if we should `continue` here? But probably fine if its just for alchemy!

File: tokenprocessing/process.go [hunk 3]
```diff
@@ -243,6 +246,272 @@ func processOwnersForUserTokens(mc *multichain.Provider, queries *coredb.Queries
...
+			newTokens, err := mc.SyncTokensByUserIDAndTokenIdentifiers(c, userID, util.MapWithoutError(tokens, func(t alchemyTokenIdentifiers) persist.TokenUniqueIdentifiers {
+				return persist.TokenUniqueIdentifiers{
+					Chain:           t.Chain,
+					ContractAddress: t.ContractAddress,
+					TokenID:         t.TokenID,
+					OwnerAddress:    t.OwnerAddress,
+				}
+			}))
+			if err != nil {
+				util.ErrResponse(c, http.StatusInternalServerError, err)
```
Comment (jarrel-b) [inline tokenprocessing/process.go:452]: Would we want to `continue` here so we could process other users?

**Review by jarrel-b — APPROVED** [review]
Looks awesome!

**Pull request merged**